File: Trabalhos - 05 - Quinto Periodo/GBC055 - POO2/AulaPratica06/src/aulapratica06/AulaPratica06.java
/*
Programa: Aula Pratica 06 - POO2
Alunos : Adriano Araujo Martins de Resende 11611BCC015
         Marcelo Mendonca Borges           11611BCC020
Data: 10/05/2018
*/

package aulapratica06;

import java.sql.Date;

public class AulaPratica06
{
    public static void main(String[] args)
    {
        Account ac = new Account(0);
        BalanceCalculator ba = new BalanceCalculatorA();
        BalanceAdapter ad = new BalanceAdapter(ba);
        BalanceResolver br = ad;
        Date date = Date.valueOf("2018-05-10");
        
        ad.balanceFor(ac, date);
    }
    
}

File: Trabalhos - 03 - Terceiro Periodo/GBC033 - PF/AulaPratica06.hs
------------------------------------
-- Programa: Aula Pratica 06      --
-- Aluno: Marcelo Mendonca Borges --
-- Data: 02/06/2017               --
------------------------------------

produto :: Num t => [t] -> t
produto [x] = x
produto (x:xs) = x * produto xs

iguais :: Eq t => t -> t -> t -> Bool
iguais n m p = (n == m) && (m == p)

tamanho :: [t] -> Int
tamanho [] = 0
tamanho (x:xs) = 1 + tamanho xs

f :: Num a => a -> [t] -> [t] -> (a, [t])
f x y z = (x + 1, y ++ z)

rep :: (Num a, Eq a)  => a -> b -> [b]

rep 0 ch = []
rep n ch = ch : rep (n-1) ch
